hip/components/atomic.hip.hpp

```diff
@@ -52,8 +52,8 @@ __forceinline__ __device__ thrust::complex<float> atomic_add(
 {
     hipComplex *addr = reinterpret_cast<hipComplex *>(address);
     // Separate to real part and imag part
-    auto real = atomic_add(&(addr->x), val.real());
-    auto imag = atomic_add(&(addr->y), val.imag());
+    auto real = atomic_add(static_cast<float *>(&(addr->x)), val.real());
+    auto imag = atomic_add(static_cast<float *>(&(addr->y)), val.imag());
```

Comment:
I really hate that we have to do this cast since thrust::complex gives no reference access to the data members. Or does anybody see a more sensible approach to accessing them?
This feels like borderline undefined behavior

Reply:
The thrust part should be kind of fine since we reinterpret it to `hipComplex`. `addr->x` should have direct access (since we access a member variable and not a function).